server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1809,4 +1809,18 @@ public String toString(Params params) {
         }
     }
 
+    /**
+     * Returns a minimal source that discards information that aren't required by shard-level search requests.
+     * The purpose is to minimize the memory usage and serialization cost of shard-level search requests.
+     */
+    public SearchSourceBuilder minimalSourceForShardRequests() {
+        if (pointInTimeBuilder != null || collapse != null) {
+            final SearchSourceBuilder source = shallowCopy();
+            source.pointInTimeBuilder(null);
+            source.collapse(null);
```

Comment:
The collapse builder is used to build the search context so we should not remove it ?

Reply:
Good catch. Fixed in https://github.com/elastic/elasticsearch/pull/80634/commits/d99725efd5f247c3ffe1e6b5c8f9e679ef43129e